src/dynamic_honey_badger/dynamic_honey_badger.rs

```diff
@@ -258,7 +259,12 @@ where
         hb_step: honey_badger::Step<InternalContrib<C, N>, N>,
     ) -> Result<Step<C, N>> {
         let mut step: Step<C, N> = Step::default();
-        let output = step.extend_with(hb_step, |hb_msg| Message::HoneyBadger(self.era, hb_msg));
+        let to_fault = |hb_fault: Fault<N, honey_badger::FaultKind>| {
+            Fault::new(hb_fault.node_id, FaultKind::HbFault(hb_fault.kind))
```

Comment:
Maybe this functionality should be extracted into a `Fault::map` convenience method, so this line would be easier on the eye:

```rust
hb_fault.map(FaultKind::HbFault)
```